### PR TITLE
Add CPU usage test for the optimization of port-buffer-drop counterpoll

### PR DESCRIPTION
**Pull request by JibinBao**
### Description of PR

There is a performance optimization for port buffer drop counter. Adjust port-buffer-drop counter default interval from 60000ms to 10000ms.

Test is skipped on 201811, 201911 and 202012. It should work on 202106 and later images
Test will check DUT memory usage and process cpu usage are within threshold for 60s.

- Disable all counterpoll types except for the tested one
- Collect memory and CPU usage for 60 secs
- Compare the memory usage with the memory threshold
- Compare the average cpu usage with the cpu threshold for the specified progress(for nvidia, the progress is sx_sdk)
- Restore counterpoll interval and status to state before the test

Summary:
Fixes # (issue)
Regarding PRs: 
https://github.com/Azure/sonic-utilities/pull/1683
https://github.com/Azure/sonic-swss/pull/1792

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911

#### What is the motivation for this PR?
Add new test for the optimization of port-buffer-drop
PR: https://github.com/Azure/sonic-utilities/pull/1683

#### How did you do it?
Compare the memory usage and cpu usage with the threshold

#### How did you verify/test it?
Run the test with the specified cpu usage threshold of port_buffer_drop_cpu_usage_threshold.

#### Any platform specific information?
No
#### Supported testbed topology if it's a new test case?
Any

### CODE REVIEW

**Review by roysr-nv — DISMISSED** [review]

**Comment by liat-grozovik** [comment]
@JibinBao please fix new alerts

**Comment by JibinBao** [comment]
The lgtm's warning is a false positive. I have changed it as lgtm's prompt. But it still has the warning.
Maybe it will take effect after merge the code. 
I see there is some prompt like " If you make changes to the suppression of alerts in a pull request, please note that theses changes will only be effective once that pull request is merged. "

**Review by roysr-nv — DISMISSED** [review]

**Comment by liat-grozovik** [comment]
@neethajohn kindly reminder to review and signoff this PR

**Comment by JibinBao** [comment]
@neethajohn Can you help review this PR?

File: tests/platform_tests/counterpoll/counterpoll_helper.py
Comment (wangxin) [inline tests/platform_tests/counterpoll/counterpoll_helper.py:23]: This indention is 8 whitespaces. It should be 4. This could raise exception.

Comment (JibinBao) [reply tests/platform_tests/counterpoll/counterpoll_helper.py]: Fixed. Thank you @wangxin

**Comment by JibinBao** [comment]
For lgtm alerts, I have added "lgtm [py/unused-import]". It seems only when the code is merged it can take effect.

**Review by wangxin — APPROVED** [review]

**Comment by bingwang-ms** [comment]
@JibinBao I opened another PR to improve this test case https://github.com/sonic-net/sonic-mgmt/pull/6851. Please help review. Thanks

**Pull request merged**
